### PR TITLE
Feat/logout

### PR DESCRIPTION
**Pull request by JeanBarriere**
- [x] add a `DELETE` request to `/user` to delete the user cookie and therefore logout the user

**Why is this needed ?**
- To have access to the cookie deletion from another domain than `api.hub.asyncy.com`
- To prevent platforms updates in case we update the login method (today it deletes a cookie, but it may change in the future)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
 target/
+hub.iml
```
Comment (judepereira) [inline .gitignore]:
```suggestion
*.iml
```

File: src/main/java/com/asyncy/apis/hub/controllers/UserController.java [hunk 2]
```diff
@@ -15,4 +20,17 @@
     public User get(final HttpServletRequest request) {
         return fetchUser(request);
     }
+
+    @RequestMapping(path="/user", method = DELETE)
+    public void invalidate(final HttpServletRequest request, final HttpServletResponse response) {
+        UserCreds owner = fetchOwner(request);
+        if (owner.getUuid() != null && !owner.getUuid().equals("")) {
```
Comment (judepereira) [inline src/main/java/com/asyncy/apis/hub/controllers/UserController.java]: Handy one liner to do the same.

```suggestion
        if (!"".equals(owner.getUuid())) {
```

Comment (judepereira) [reply src/main/java/com/asyncy/apis/hub/controllers/UserController.java]: Btw, even more handier is to use StringUtils.isNotBlank() from the Apache commons lang library. https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isNotBlank-java.lang.CharSequence-

File: .gitignore (hunk 1 above)
Comment (JeanBarriere) [reply .gitignore]: well the project name is hub so no need for that :)

File: src/main/java/com/asyncy/apis/hub/controllers/UserController.java (hunk 2 above)
Comment (JeanBarriere) [reply src/main/java/com/asyncy/apis/hub/controllers/UserController.java]: Indeed

File: .gitignore (hunk 1 above)
Comment (judepereira) [reply .gitignore]: Yes, but your name for the project is `hub`. Mine is `hub-api`. Somebody else can name it `Asyncy Hub API`. It's in our best interest to add `*.iml` altogether.

**Pull request merged**